### DISCUSSION TITLE
GO CAM Programmatic access

### DISCUSSION THREAD

**Discussion by arihunter** [geneontology/helpdesk#450] [General]
Hi, I would like to fetch the relation types and details of model components like GO_ID, Gene Product, and References through an API. I went through the provided rest API documentation. The links provided (ex - http://model.geneontology.org/5b91dbd100001993/5b91dbd100001994) don't provide metadata of model components. 
Any help to get this metadata will be appreciated.

**Comment by suzialeksander** [comment]
Thanks for contacting the GO Helpdesk, @arihunter.  I'm tagging @kltm to answer this.

**Comment by kltm** [comment]
@arihunter We are currently undergoing an API refactoring project and are beginning to revisit how we present access to GO-CAM information. Unfortunately, as we roll out our new devops, we have temporarily lost access to the OpenAPI online documentation (known bug to be remedied in the very near future). Until then, I'm going to loop in @sierra-moxon, who may be able to address whether this information is readily available in the currently deployed iteration of the API.

**Reply by arihunter** [reply]
thanks, @kltm for the response, it will be great if some resolution is brought to the issue.

**Reply by sierra-moxon** [reply]
Hi @arihunter - 

Part of our API refactoring project is to consolidate two APIs ([api.geneontology.org](https://api.geneontology.org/) and our internal GO-CAM API (that does have endpoints to query annotations by model id) to one, public-facing API. 

It would be really helpful to understand if you need more than `GO_ID, Gene Product, and References` in an endpoint queryable by model id?  I'd be happy to do some endpoint development to your use case in the new consolidated API which coincidentally already has some endpoints for accessing annotation metadata by model id (but not exactly those three items). 

And @kltm mentioned that watching go-announce, or joining the go-friends mail list would be helpful in getting notifications of when the migration is complete and these new endpoints go live.

**Reply by arihunter** [reply]
hi @sierra-moxon, thanks for replying. For each GO-CAM model, I require the type of relation between two annotations and details like (GO_ID, gene product, references) for each annotation. This should happen for all the relations existing in the respective GO-CAM model.

**Comment by arihunter** [comment]
Hi @sierra-moxon @kltm , as this is related to a project of mine and has a approaching deadline , Can you provide any resolution to the query

**Comment by kltm** [comment]
@arihunter The timeline for any modifications to the API would be in months, at the earliest. @sierra-moxon will be back in the office in early August and can follow up with any specific GO API questions you might have then.

**Comment by kltm** [comment]
@arihunter I had a chance to talk to @sierra-moxon and this is not a route that exists in the API. We have it slated for discussion for future iterations of the API https://github.com/geneontology/go-fastapi/issues/61.
You may be able to get what you want from either the TTL (available in GitHub) or JSON (soon available via the API once we go live with the current code update).
